client/components/room/Room.jsx

```diff
@@ -23,6 +25,8 @@ class Room extends React.Component {
       console.log(`${room} is full`);
       this.setState({ redirect: true });
     });
+
+    !this.props.userName ? this.props.addUserName(generateSillyName()) : null;
```

Comment:
Took me a minute to realize what this ternary statement was doing, but it is getting late, lol. Pretty slick way to conditionally generate the username.